### DISCUSSION TITLE
I can't install opentimeline io using pip

### DISCUSSION THREAD

**Discussion by MichaelPlug** [AcademySoftwareFoundation/OpenTimelineIO#1342] [Q&A] [answered]
Hi, i'm tring to install opentimelineio on my macOS 12.4 using pip and it fail building building wheels 

  ```
      File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/subprocess.py", line 373, in check_call
          raise CalledProcessError(retcode, cmd)
      subprocess.CalledProcessError: Command '['make', 'install', '-j4']' returned non-zero exit status 2.
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for opentimelineio
  Running setup.py clean for opentimelineio
Failed to build opentimelineio
Installing collected packages: opentimelineio
  Running setup.py install for opentimelineio ... error
  error: subprocess-exited-with-error
  
  × Running setup.py install for opentimelineio did not run successfully.
  │ exit code: 1
  ╰─> [945 lines of output]

```

```
        File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/subprocess.py", line 373, in check_call
          raise CalledProcessError(retcode, cmd)
      subprocess.CalledProcessError: Command '['make', 'install', '-j4']' returned non-zero exit status 2.
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: legacy-install-failure

× Encountered error while trying to install package.
╰─> opentimelineio

note: This is an issue with the package mentioned above, not pip.
hint: See above for output from the failure.

```

**Comment by jminor** [comment, marked as answer]
Looking at the long output (which I see in email, but not on the GitHub web site for some reason) I see a couple of clues that might help:
- The command `python3 -m pip install --upgrade -pip` should be `python3 -m pip install --upgrade pip` (without the "-" before "pip")
- For some unknown reason, pip is choosing to build OTIO instead of using a pre-built wheel
- The build is failing because you don't have cmake installed.

Also, I believe that Apple no longer ships Python since macOS 12.4, but you seem to have Python installed in "/Library/Frameworks/Python.framework/Versions/3.9" which implies that you installed Python via the standard Python installer. That may be a configuration that we haven't tested before. We generally recommend using either miniconda or pyenv to install an isolated Python environment. I don't know if that will make a difference, but it could be something to try.

It is possible that upgrading pip (the first bullet point above) could resolve the wheel issue. If that works and you're able to get a pre-built wheel that would be ideal, since it would skip the other issues.

**Reply by MichaelPlug** [reply]
> * The command `python3 -m pip install --upgrade -pip` should be `python3 -m pip install --upgrade pip` (without the "-" before "pip")
> * For some unknown reason, pip is choosing to build OTIO instead of using a pre-built wheel
> * The build is failing because you don't have cmake installed.

I installed cmake and I upgrated pip and now the output is the short one in the last message. Thank you for your suggestion.

**Reply by JeanChristopheMorinPerso** [reply]
>For some unknown reason, pip is choosing to build OTIO instead of using a pre-built wheel

@MichaelPlug Is your mac an arm64 (Mac M1)? Because that would explain why pip build from source instead of installing our pre-build wheels.

**Comment by AustinWongParker** [comment]
I think I'm having a similar issue but not on macOS. Going through the installation for otio (windows machine) and am getting the [following error](https://pastebin.pl/view/69b5b801).
I've already upgraded pip to 22.1.2, installed cmake, and just cloned otio from the repo.
This error comes up when I run: `python -m pip install .`

**Comment by meshula** [comment]
@AustinWongParker I notice in your log that cmake is trying to build in a mingw environment, and failing with an nmake generator step. We've only tested building otio on windows with a "traditional" msvc environment. Are you running from a visual studio command prompt when you run pip install?

**Reply by AustinWongParker** [reply]
@meshula I'm running from a cygwin64 terminal to mimic the linux terminal. I'll try running it with a msvc compiler

**Reply by AustinWongParker** [reply]
Got it working by using a different terminal. After installing the dependencies via pip, I'm confused on where this **edl** file is for _otioview path/to/your/file.edl_

**Reply by apetrynet** [reply]
Hi! 
Please excuse me if I misunderstood your question.

The edl file is usually a file you produce from an nle like premier or receive from a client for instance.
Otioview will visualize the timeline based on the edl file or any other supported file format. 

There should be some examples in the tests/sample_data folder.

**Reply by AustinWongParker** [reply]
Oh, gotcha. Yeah I misunderstood - that clears it up. Thank you

**Comment by jminor** [comment]
@MichaelPlug were you able to get OpenTimelineIO installed and running?

**Reply by MichaelPlug** [reply]
Yes, thank you
